### DISCUSSION TITLE
Interacting with streaming microphone in python

### DISCUSSION THREAD

**Discussion by PA-Alex** [deepgram/community#724] [General help] [answered]
Hello. 

I must be doing something very silly, but I can't seem to be able to interact with the microphone streaming setup in Python. 
I'm using github.com/deepgram/deepgram-python-sdk/tree/main/examples/streaming/microphone/main.py

What I've done so far:
- Taken the microphone example and converted it to a class where the __init__(self) sets up the dg_connection and registers the LiveTranscriptionEvents. All other functions in the main example have been moved to be class functions.

- Moved the microphone.finish() and dg_connection.finish() to a close function as to not to loose the stream at the end of __init__.

- Created a function on the class [**transcribe**()] that returns the content of **utterance** once the result.is_final flag or on_utterance_end get called. 

On the main function, I create my DGMicrophone() object -> No problems here. When I call DGM.transcribe(), DGMicrophone() does not seize control back; it seems that the execution is moving on_message

Is there an easy way to setup and maintain a connection to Deepgram and allow the utterance capture text to be sent out of the class?

Alex

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment]
Did you look at the README for requirements?
https://github.com/deepgram/deepgram-python-sdk/tree/main/examples/streaming/microphone

Also, will not work on Windows environments. This is for Linux-based OS only.

**Comment by PA-Alex** [comment]
Thank you Dvontheen. 

Oh, no. It is running (it took a bit of refactoring to rewire the signals, but afterwards, it was fine).
What it was not doing was allowing me to get the utterance onto a variable outside the class. 

For the record, I found a workaround by using static variables. For whatever reason the class variables were misbehaving. 

Alex

**Reply by davidvonthenen** [reply, marked as answer]
You can use globals, but probably the preferred way would be to extend the client class.

**Reply by PA-Alex** [reply]
Thank you again! :)

Could you point me towards an example? (Or paste the bit of code that would allow me to do it?) I come from the java world and have not programmed in python for long.

Regards 
Alex
